--- tclvfs/generic/vfs.c	2009-02-26 14:08:20.000000000 +0100
+++ tclvfs-patched/generic/vfs.c	2009-02-26 14:08:12.000000000 +0100
@@ -1558,6 +1558,8 @@ VfsMatchInDirectory(
 				  TCL_EVAL_GLOBAL | TCL_EVAL_DIRECT);
 	if (returnVal != TCLVFS_POSIXERROR) {
 	    vfsResultPtr = Tcl_DuplicateObj(Tcl_GetObjResult(interp));
+        } else {
+            returnVal = TCL_OK;
 	}
 	Tcl_RestoreResult(interp, &savedResult);
 	Tcl_DecrRefCount(mountCmd);
