--- vqtcl/library/mkclvfs.tcl	2007-04-17 16:24:31.000000000 +0200
+++ vqtcl-patched/library/mkclvfs.tcl	2009-02-26 14:28:40.000000000 +0100
@@ -103,7 +103,9 @@ namespace eval ::vfs::mkcl {
     proc matchindirectory {db path actual pattern type} {
         set o {}
         if {$type == 0} { set type 20 }
-        set tag [lookUp $db $path]
+        if {[catch {lookUp $db $path} tag]} {
+            return {}
+        }
         if {$pattern ne ""} {
             set c {}
             if {[isDir $tag]} {
